services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/PortPropertiesSpec.groovy

```diff
@@ -0,0 +1,250 @@
+                new PortPropertiesDto(discoveryEnabled: true))
+        northboundV2.updatePortProperties(islToManipulate.dstSwitch.dpId, islToManipulate.dstPort,
+                new PortPropertiesDto(discoveryEnabled: true))
+
+        then: "System discovers link and link status is DISCOVERED"
+        Wrappers.wait(discoveryInterval + WAIT_OFFSET) {
+            assert islUtils.getIslInfo(islToManipulate).get().state == IslChangeType.DISCOVERED
+        }
+        Boolean testIsCompleted = true
+
+        cleanup: "Restore discovery port property on the port"
+        if (!testIsCompleted) {
+            northboundV2.updatePortProperties(islToManipulate.srcSwitch.dpId, islToManipulate.srcPort,
+                    new PortPropertiesDto(discoveryEnabled: true))
+            northboundV2.updatePortProperties(islToManipulate.dstSwitch.dpId, islToManipulate.dstPort,
+                    new PortPropertiesDto(discoveryEnabled: true))
+        }
+    }
+
+    def "Link is stop being discovered afrter disabling port discovery feature"() {
+        given: "A link with a flow on it"
+        def switchPair = getTopologyHelper().getAllNeighboringSwitchPairs().find { it.paths.size() >= 2 }
+        def flow = flowHelper.randomFlow(switchPair)
+        flowHelper.addFlow(flow)
+        def flowPath = northbound.getFlowPath(flow.id)
+        def islToManipulate = pathHelper.getInvolvedIsls(flowPath).first()
+
+        when: "Disable discovery port feature"
+        northboundV2.updatePortProperties(islToManipulate.srcSwitch.dpId, islToManipulate.srcPort,
+                new PortPropertiesDto(discoveryEnabled: false))
+
+        then: "Link state is changed to FAILED"
+        Wrappers.wait(discoveryTimeout + WAIT_OFFSET) {
+            assert islUtils.getIslInfo(islToManipulate).get().state == IslChangeType.FAILED
+        }
+
+        and: "Flow is rerouted"
+        def newFlowPath
+        Wrappers.wait(rerouteDelay + WAIT_OFFSET) {
+            newFlowPath = northbound.getFlowPath(flow.id)
+            newFlowPath != flowPath
+        }
+
+        when: "Enable discovery port property"
+        northboundV2.updatePortProperties(islToManipulate.srcSwitch.dpId, islToManipulate.srcPort,
+                new PortPropertiesDto(discoveryEnabled: true))
+
+        then: "Link state is changed to DISCOVERED"
+        Wrappers.wait(discoveryInterval + WAIT_OFFSET) {
+            assert islUtils.getIslInfo(islToManipulate).get().state == IslChangeType.DISCOVERED
+        }
+        Boolean testIsCompleted = true
+
+        and: "Flow is not rerouted"
+        northbound.getFlowPath(flow.id) == newFlowPath
+
+        cleanup: "Restore discovery port property on the port"
+        if (!testIsCompleted) {
+            northboundV2.updatePortProperties(islToManipulate.srcSwitch.dpId, islToManipulate.srcPort,
+                    new PortPropertiesDto(discoveryEnabled: true))
+        }
+    }
```

Comment:
I understand the point of this test, but I don't understand what flow is doing here. Looks kinda unrelated. It's pretty obvious that the flow is rerouted when ISL fails so I'm not sure what did you try to show here. You mean the event of ISL failure is being 'different' in this case? That's veeery unlikely, I don't think we should test it here

Reply:
test was refactored, flow was removed